File: somalin/somalin.py
#!/usr/bin/env python3

"""Somar números na linha de comando"""

import sys

total = 0.0
for arg in sys.argv[1:]:
    try:
        x = float(arg)
    except ValueError:
        print('*** número inválido:', arg)
        continue
    total += x

print('Total:', total)

File: cli/somalin/somalin.go
// Somar números na linha de comando

package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	total := 0.0
	for _, arg := range os.Args[1:] {
		x, err := strconv.ParseFloat(arg, 64)
		if err != nil {
			fmt.Println("*** número inválido:", arg)
			continue
		}
		total += x
	}
	fmt.Println("Total:", total)
}
